test/spec/test-common.js

```diff
@@ -304,6 +304,35 @@ describe('tus', () => {
       expect(options.onAfterResponse).toHaveBeenCalled()
     })
 
+    it('should invoke the onUploadUrlAvailable callback', async () => {
+      const testStack = new TestHttpStack()
+      const file = getBlob('hello world')
+      const options = {
+        httpStack: testStack,
+        uploadUrl: 'http://tus.io/uploads/foo',
+        onUploadUrlAvailable: waitableFunction('onUploadUrlAvailable'),
+        onSuccess: waitableFunction('onSuccess'),
+      }
+
+      const upload = new tus.Upload(file, options)
+      upload.start()
+
+      const req = await testStack.nextRequest()
+      expect(req.url).toBe('http://tus.io/uploads/foo')
+      expect(req.method).toBe('HEAD')
+
+      req.respondWith({
+        status         : 204,
+        responseHeaders: {
+          'Upload-Offset': 11,
+          'Upload-Length': 11,
+        },
+      })
+
+      await options.onSuccess.toBeCalled
+      expect(options.onUploadUrlAvailable).toHaveBeenCalled()
```

Comment:
Can we move this assertion before `await options.onSuccess.toBeCalled` so we know that the callback is invoked directly after the HEAD response? Or does this asynchronous behavior interfere with this?

Reply:
See comment below.